pkg/util/stop/stopper.go

```diff
@@ -474,9 +474,9 @@ func (s *Stopper) RunAsyncTaskEx(ctx context.Context, opt TaskOpts, f func(conte
 	var sp *tracing.Span
 	switch opt.SpanOpt {
 	case FollowsFromSpan:
-		ctx, sp = tracing.EnsureForkSpan(ctx, s.tracer, opt.TaskName)
+		ctx, sp = tracing.ForkSpan(ctx, opt.TaskName)
 	case ChildSpan:
-		ctx, sp = tracing.EnsureChildSpan(ctx, s.tracer, opt.TaskName)
+		ctx, sp = tracing.ChildSpan(ctx, opt.TaskName)
```

Comment:
These changes are slightly more interesting to me. It seems like a good idea to force async tasks to have spans regardless of the call-site. It seems like the overhead might be worth it in these cases, but perhaps my intuition is wrong.

Reply:
It might be slightly misleading that they're called "async tasks", as they are in fact used for spawning short-lived request goroutines in the hot path. For example `DistSender` does this, and I think `kvstreamer` too. Besides, if async tasks have spans, what benefit do we derive from this? I think back in the day we were hoping that we'd use the async span registry to look at what the goroutines are doing, etc, but nowadays I am more inclined to believe that execution traces or eBPF or some combination thereof will fill our observability needs better.

Reply:
I think this was introduced here https://github.com/cockroachdb/cockroach/commit/2ad2bee257e78970ce2c457ddd6996099ed6727a - note that there isn't strong motivation other than the intuition that you had - but since we use these for short-running frequent tasks a lot of the time, it makes no sense to have trace spans here. And even if it made sense, we wouldn't be making them visible anywhere, so from that point of view it could only make sense to keep doing this if we were also gearing up to cash in on this information. It doesn't look like anybody is, so I'm very comfortable removing this.